subsystem/core/src/main/java/org/wildfly/extension/camel/service/CamelEndpointDeployerService.java

```diff
@@ -0,0 +1,497 @@
+            endpointHttpHandler.service(getServletContext(), req, res);
+        }
+
+        public void setEndpointHttpHandler(EndpointHttpHandler endpointHttpHandler) {
+            this.endpointHttpHandler = endpointHttpHandler;
+        }
+
+    }
+
+    static class UrlPattern {
+
+        private final String source;
+
+        public UrlPattern(String source) {
+            super();
+            this.source = source;
+        }
+
+        public String relativize(String path) {
+            final StringTokenizer patternTokenizer = new StringTokenizer(source, "/", true);
+            final StringTokenizer pathTokenizer = new StringTokenizer(path, "/", true);
+            /* skip the common segments */
+            while (pathTokenizer.hasMoreTokens()) {
+                if (!patternTokenizer.hasMoreTokens()) {
+                    return null;
+                }
+                final String patternToken = patternTokenizer.nextToken();
+                final String pathToken = pathTokenizer.nextToken();
+                final boolean isSlash = patternToken.equals("/");
+                if (isSlash != pathToken.equals("/")) {
+                    return null;
+                } else if (!isSlash) {
+                    if ("*".equals(pathToken)) {
+                        continue;
+                    } else {
+                        final StringTokenizer segmentTokenizer = new StringTokenizer(patternToken, "*", true);
+                        final StringBuilder segmentPatternBuilder = new StringBuilder(patternToken.length());
+                        while (segmentTokenizer.hasMoreTokens()) {
+                            final String segmentToken = segmentTokenizer.nextToken();
+                            if ("*".equals(segmentToken)) {
+                                segmentPatternBuilder.append(".*");
+                            } else {
+                                segmentPatternBuilder.append(Pattern.quote(segmentToken));
+                            }
+                        }
+                        final Pattern pat = Pattern.compile(segmentPatternBuilder.toString());
+                        if (!pat.matcher(pathToken).matches()) {
+                            return null;
+                        }
+                    }
+                }
+            }
+            final StringBuilder result = new StringBuilder();
+            while (patternTokenizer.hasMoreTokens()) {
+                result.append(patternTokenizer.nextToken());
+            }
+            if (result.length() == 0) {
+                return "";
+            } else if (result.charAt(0) != '/') {
+                return "/" + result.toString();
+            } else {
+                return result.toString();
+            }
+        }
+
+    }
+
+    private static final String CATCH_ALL_ENDPOINT_URI_PREFIX = "///*";
+
+    private static final String CATCH_ALL_PREFIX = "/*";
```

Comment:
Not keen on the `CATCH_ALL` prefix but can't think of an decent alternative at the moment.

Reply:
`MATCH_ALL...` perhaps better?

Reply:
`MATCH_ALL` sounds better :+1:

Reply:
OK, let me rename it